# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/msaa/AccessibleWrap.h
# Commit: ffa79d70878f
# Full Hash: ffa79d70878f731b772ecf7a6d26e24eb8eaa5cb
# Author: Trevor Saunders <tbsaunde@tbsaunde.org>
# Date: 2015-05-06 20:26:41
# Regressor Bug: 606080
# File Overlap Count: 1
# Description:
#   bug 606080 - on windows give accessibles a unique 32 bit id r=surkov
# ==============================================================================

diff -r 28c2b07d197c -r ffa79d70878f accessible/windows/msaa/AccessibleWrap.h
--- a/accessible/windows/msaa/AccessibleWrap.h	Mon Apr 13 18:30:07 2015 -0400
+++ b/accessible/windows/msaa/AccessibleWrap.h	Tue Apr 14 12:45:43 2015 -0400
@@ -15,6 +15,7 @@
 #include "ia2AccessibleHyperlink.h"
 #include "ia2AccessibleValue.h"
 #include "mozilla/a11y/ProxyAccessible.h"
+#include "mozilla/a11y/IDSet.h"
 
 #ifdef __GNUC__
 // Inheriting from both XPCOM and MSCOM interfaces causes a lot of warnings
@@ -33,8 +34,7 @@
                        public ia2AccessibleValue
 {
 public: // construction, destruction
-  AccessibleWrap(nsIContent* aContent, DocAccessible* aDoc) :
-    Accessible(aContent, aDoc) { }
+  AccessibleWrap(nsIContent* aContent, DocAccessible* aDoc);
 
   // nsISupports
   NS_DECL_ISUPPORTS_INHERITED
@@ -152,6 +152,7 @@
 
   // Accessible
   virtual nsresult HandleAccEvent(AccEvent* aEvent);
+  virtual void Shutdown() override;
 
   // Helper methods
   static int32_t GetChildIDFor(Accessible* aAccessible);
@@ -176,7 +177,11 @@
   static IDispatch* NativeAccessible(Accessible* aAccessible);
 
 protected:
-  virtual ~AccessibleWrap() { }
+  virtual ~AccessibleWrap();
+
+#ifdef _WIN64
+  uint32_t mID;
+#endif
 
   /**
    * Creates ITypeInfo for LIBID_Accessibility if it's needed and returns it.
@@ -185,6 +190,9 @@
 
   static ITypeInfo* gTypeInfo;
 
+#ifdef _WIN64
+  static IDSet sIDGen;
+#endif
 
   enum navRelations {
     NAVRELATION_CONTROLLED_BY = 0x1000,